components/calendar/calendar.js

```diff
@@ -663,87 +707,122 @@ class Calendar extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 			} case keyCodes.PAGEUP: {
 				const diff = getNumberOfDaysToSameWeekPrevMonth(this._shownMonth, this._shownYear);
 
-				this._focusDate.setDate(this._focusDate.getDate() - diff);
+				const possibleFocusDate = new Date(this._focusDate);
+				possibleFocusDate.setDate(possibleFocusDate.getDate() - diff);
 				this._keyboardTriggeredMonthChange = true;
 
 				// handle when current month has more weeks than previous month and page up pressed from last week
-				if (this._focusDate.getMonth() === this._shownMonth) this._focusDate.setDate(this._focusDate.getDate() - 7);
+				if (possibleFocusDate.getMonth() === this._shownMonth) possibleFocusDate.setDate(possibleFocusDate.getDate() - 7);
 
 				this._updateShownMonthDecrease(true, true);
+				await this.updateComplete;
+				this._updateFocusDateDependentOnDisabled(possibleFocusDate);
+				if (this._focusDate) this._focusDateAddFocus();
+				else this.focus();
 				preventDefault = true;
 				break;
 			} case keyCodes.PAGEDOWN: {
 				const nextMonth = getNextMonth(this._shownMonth);
 				const diff = getNumberOfDaysToSameWeekPrevMonth(nextMonth, this._shownYear);
 
-				this._focusDate.setDate(this._focusDate.getDate() + diff);
+				const possibleFocusDate = new Date(this._focusDate);
+
+				possibleFocusDate.setDate(possibleFocusDate.getDate() + diff);
 				this._keyboardTriggeredMonthChange = true;
 
 				// handle when current month has more weeks than next month and page down pressed from last week
-				const newFocusDateMonth = this._focusDate.getMonth();
+				const newFocusDateMonth = possibleFocusDate.getMonth();
 				if ((newFocusDateMonth - this._shownMonth) > 1
 					|| (newFocusDateMonth === 1 && this._shownMonth === 11)
 				) {
-					this._focusDate.setDate(this._focusDate.getDate() - 7);
+					possibleFocusDate.setDate(possibleFocusDate.getDate() - 7);
 				}
 				this._updateShownMonthIncrease(true, true);
+				await this.updateComplete;
+
+				this._updateFocusDateDependentOnDisabled(possibleFocusDate, true);
+				if (this._focusDate) this._focusDateAddFocus();
+				else {
+					const buttons = this.shadowRoot.querySelectorAll('d2l-button-icon');
+					if (buttons && buttons.length > 1) buttons[1].focus();
+				}
 				preventDefault = true;
 				break;
 			}
 		}
 
-		if (numDaysChange) {
-			this._isInitialFocusDate = false;
-			this._updateFocusDateOnKeyDown(numDaysChange);
-		}
-
 		if (preventDefault) {
 			e.preventDefault();
 			e.stopPropagation();
 		}
+
+		if (numDaysChange) {
```

Comment:
I moved the functionality of `_updateFocusDateOnKeyDown` into here since it was not used anywhere else and then tweaked a little to deal with disabled focus dates.